src/ScopeProvider.tsx

```diff
@@ -18,70 +19,90 @@ export const ScopeProvider = ({
+          read(get, opts) {
+            return anAtom.read.call(
+              this,
+              (a) => get(getAtom(this, anAtom, a)),
+              opts,
+            );
+          },
+        }),
+        ...('write' in anAtom && {
+          write(get, set, ...args) {
+            return anAtom.write.call(
+              this,
+              (a) => get(getAtom(this, anAtom, a)),
+              (a, ...v) => set(getAtom(this, anAtom, a), ...v),
+              ...args,
+            );
+          },
+        }),
+      };
+      return scopedAtom;
+    };
+
+    const getScopedAtom: GetScopedAtom = (anAtom) => {
+      let scopedAtom = mapping.get(anAtom);
+      if (!scopedAtom) {
+        scopedAtom = atomSet.has(anAtom)
+          ? createScopedAtom(anAtom as unknown as AnyWritableAtom, false)
+          : createScopedAtom(anAtom as unknown as AnyWritableAtom, true);
+        mapping.set(anAtom, scopedAtom);
       }
-      return getScopedAtom(target);
+      return scopedAtom as typeof anAtom;
     };
-    const scopedAtom: typeof anAtom = {
-      ...anAtom,
-      ...('read' in anAtom && {
-        read(get, opts) {
-          return anAtom.read.call(
-            this,
-            (a) => get(getAtom(this, anAtom, a)),
-            opts,
-          );
-        },
-      }),
-      ...('write' in anAtom && {
-        write(get, set, ...args) {
-          return anAtom.write.call(
-            this,
-            (a) => get(getAtom(this, anAtom, a)),
-            (a, ...v) => set(getAtom(this, anAtom, a), ...v),
-            ...args,
-          );
-        },
-      }),
+
+    const patchedStore: typeof store = {
+      ...store,
+      get: (anAtom, ...args) => store.get(getScopedAtom(anAtom), ...args),
+      set: (anAtom, ...args) => store.set(getScopedAtom(anAtom), ...args),
+      sub: (anAtom, ...args) => store.sub(getScopedAtom(anAtom), ...args),
     };
-    return scopedAtom;
-  };
 
-  const getScopedAtom: GetScopedAtom = (anAtom) => {
-    let scopedAtom = mapping.get(anAtom);
-    if (!scopedAtom) {
-      scopedAtom = atomSet.has(anAtom)
-        ? createScopedAtom(anAtom as unknown as AnyWritableAtom, false)
-        : createScopedAtom(anAtom as unknown as AnyWritableAtom, true);
-      mapping.set(anAtom, scopedAtom);
-    }
-    return scopedAtom as typeof anAtom;
+    return [
+      patchedStore,
+      getScopedAtom,
+      store,
+      getParentScopedAtom,
+      atomSet,
+    ] as const;
   };
 
-  const store = useStore();
-  const patchedStore: typeof store = {
-    ...store,
-    get: (anAtom, ...args) => store.get(getScopedAtom(anAtom), ...args),
-    set: (anAtom, ...args) => store.set(getScopedAtom(anAtom), ...args),
-    sub: (anAtom, ...args) => store.sub(getScopedAtom(anAtom), ...args),
-  };
+  const [state, setState] = useState(initialize);
+  if (
+    store !== state[2] ||
+    getParentScopedAtom !== state[3] ||
+    !isEqualSet(atomSet, state[4])
+  ) {
+    setState(initialize);
```

Comment:
Cool, TIL this usage.

Reply:
This technique is now described in the official docs: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes